### PR TITLE
RD-5134 Show only available workflows in Deployment Actions and Execute Workflow popup menus

### PR DESCRIPTION
**Pull request by qooban**
## Description

Each `workflow` resource (from `GET /deployments` API) has now `is_available` field provided. This PR uses that field to hide not availble workflows in Deployment Actions and Execute Workflow popup menus. As right now the set of visible workflows is not static for specific deployment, data polling had to be turned on for Deployment Action Buttons widget.

More info: RD-5134

## Screenshots / Videos

https://user-images.githubusercontent.com/5202105/176164402-cb1ad005-5204-4db0-a39d-b3f10d496e4e.mp4

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
1. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3002/) - failed Deployments widget test and Spire Manager widget test
2. [Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/3003/) - failed on known issues (Getting Started modal test and Blueprints Catalog test), so I'm considering as PASSED

## Documentation
Deployment Action Buttons widget description update - https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2235

### CODE REVIEW

File: test/cypress/integration/widgets/deployment_actions_spec.ts [hunk 1]
```diff
@@ -60,6 +60,20 @@ describe('Deployment Action Buttons widget', () => {
             cy.wait('@setSite');
             cy.get('.modal').should('not.exist');
         });
+
+        it('should show only available workflows and actions', () => {
+            cy.contains('button', 'Deployment actions').click();
```
Comment (kubama) [inline test/cypress/integration/widgets/deployment_actions_spec.ts:65]: `clickButton` command can be used here

**Review by kubama — DISMISSED** [review]

File: widgets/common/src/deployments/DeploymentActionsMenu.tsx [hunk 2]
```diff
@@ -11,24 +13,40 @@ export const actions = Object.freeze({
...
     { name: actions.manageLabels, icon: 'tags', permission: 'deployment_create' },
-    { name: actions.uninstall, icon: 'recycle', permission: 'execution_start' },
+    { name: actions.uninstall, icon: 'recycle', permission: executeWorkflowPermission },
     { name: actions.delete, icon: 'trash alternate', permission: 'deployment_delete' },
     { name: actions.forceDelete, icon: 'trash', permission: 'deployment_delete' }
 ];
 
+function isAvailable(item: MenuItem, workflows: Workflow[]) {
+    if (item.permission === executeWorkflowPermission) {
+        const workflow = find(workflows, { name: item.name });
```
Comment (NorbertAmartus) [inline widgets/common/src/deployments/DeploymentActionsMenu.tsx:31]: What do you think about using native `Array.prototype.find` function?
(as suggested by [UI Code Style Guidelines](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435388492/UI+Code+Style#SHOULD%3A-Use-built-in-JS-functions-instead-of-utilities-functions-from-external-packages))

```suggestion
        const workflow = workflows.find(workflow => workflow.name === item.name);
```

File: widgets/common/src/executeWorkflow/WorkflowsMenu.tsx [hunk 3]
```diff
@@ -1,130 +1,111 @@
...
+const {
+    Basic: { Accordion, Menu, Popup, PopupMenu },
+    i18n
+} = Stage;
 
-function filterWorkflows(workflows) {
+function filterWorkflows(workflows: Workflow[]) {
     const updateWorkflow = 'update';
-    return _.filter(workflows, workflow => workflow.name !== updateWorkflow);
+    return _.filter(workflows, workflow => workflow.is_available && workflow.name !== updateWorkflow);
```
Comment (NorbertAmartus) [inline widgets/common/src/executeWorkflow/WorkflowsMenu.tsx:13]: What do you think about explicit imports of the lodash functions? 🤔 
(as suggested in our [Refactors Guidelines](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/2148073473/Ongoing+Refactors+in+Cloudify+UI))

From what I can see, the following lodash functions are being used inside this file:
- `filter`
- `capitalize`
- `lowerCase`
- `map`
- `sortBy`
- `chain`
- `size`

So the corresponding import line, may look like this:
```javascript
import { capitalize, chain, filter, lowerCase, size, sortBy, map } from 'lodash';
```

File: widgets/managers/src/ExecuteWorkflowIcon.tsx [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
-    onClick?: (workflow: Parameters<ComponentProps<typeof WorkflowsMenu>['onClick']>[0]) => void;
-    workflows: ComponentProps<typeof WorkflowsMenu>['workflows'];
+    onClick?: WorkflowsMenuProps['onClick'];
+    workflows: WorkflowsMenuProps['workflows'];
 }) {
-    return !_.isEmpty(workflows) ? (
-        <WorkflowsMenu workflows={workflows} onClick={workflow => onClick(workflow)} />
-    ) : null;
+    return !_.isEmpty(workflows) ? <WorkflowsMenu workflows={workflows} onClick={onClick} /> : null;
 }
```
Comment (NorbertAmartus) [inline widgets/managers/src/ExecuteWorkflowIcon.tsx:13]: What do you think about explicit imports of the lodash functions?
(as suggested in our [Refactors Guidelines](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/2148073473/Ongoing+Refactors+in+Cloudify+UI))

```suggestion
import { isEmpty, noop } from 'lodash';
import type { WorkflowsMenuProps } from '../../common/src/executeWorkflow/WorkflowsMenu';

const WorkflowsMenu = Stage.Common.Workflows.Menu;

export default function ExecuteWorkflowIcon({
    onClick = noop,
    workflows = []
}: {
    onClick?: WorkflowsMenuProps['onClick'];
    workflows: WorkflowsMenuProps['workflows'];
}) {
    return !isEmpty(workflows) ? <WorkflowsMenu workflows={workflows} onClick={onClick} /> : null;
}
```

File: widgets/common/src/deployments/DeploymentActionsMenu.tsx (hunk 2 above)
Comment (qooban) [reply widgets/common/src/deployments/DeploymentActionsMenu.tsx]: Make sense, applied.

File: widgets/common/src/executeWorkflow/WorkflowsMenu.tsx (hunk 3 above)
Comment (qooban) [reply widgets/common/src/executeWorkflow/WorkflowsMenu.tsx]: Yep, make sense, changed to explicit imports.

File: widgets/managers/src/ExecuteWorkflowIcon.tsx (hunk 4 above)
Comment (qooban) [reply widgets/managers/src/ExecuteWorkflowIcon.tsx]: Make sense, changed to explicit imports.

**Comment by qooban** [comment]
@kubama @NorbertAmartus I've merged documentation changes (https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2235) and updated this branch, so I'm waiting for final reviews.

**Review by NorbertAmartus — DISMISSED** [review]

File: widgets/common/src/deployments/DeploymentActionsMenu.tsx [hunk 5]
```diff
@@ -1,4 +1,6 @@
+import { find } from 'lodash';
 import type { ReactNode, ComponentProps } from 'react';
```
Comment (NorbertAmartus) [inline widgets/common/src/deployments/DeploymentActionsMenu.tsx:2]: It seems like this import is not being used anymore (that's why jenkins pipelines are failing)
Here is a little fix for that 🎉 

```suggestion
import type { ReactNode, ComponentProps } from 'react';
```

**Review by NorbertAmartus — APPROVED** [review]

**Comment by qooban** [comment]
As after @kubama approval, I did only small code-style related adjustments, I'm merging this PR without his approval not to block others (as docs sync was done within this PR).

**Pull request merged**